src/payment/strategies/braintree/braintree-hosted-form.spec.ts

```diff
@@ -219,18 +219,38 @@ describe('BraintreeHostedForm', () => {
             .toHaveBeenCalledWith({ fieldType: 'cardCode' });
     });
 
-    it('notifies when card number changes', async () => {
-        const handleCardTypeChange = jest.fn();
+    describe('#cardTypeChange', () => {
+        let handleCardTypeChange: jest.Mock;
 
-        await subject.initialize({
-            ...formOptions,
-            onCardTypeChange: handleCardTypeChange,
+        beforeEach(async () => {
+            handleCardTypeChange = jest.fn();
+
+            await subject.initialize({
+                ...formOptions,
+                onCardTypeChange: handleCardTypeChange,
+            });
         });
 
-        cardFieldsEventEmitter.emit('cardTypeChange', { cards: [{ type: 'Visa' }] });
+        it('notifies when card number changes', () => {
+            cardFieldsEventEmitter.emit('cardTypeChange', { cards: [{ type: 'Visa' }] });
```

Comment:
Reading the documentation, it doesn't seem like `Visa` is a valid type. It looks like the `type` value for visa is: `visa`.